tests/sentry/utils/test_redis.py

```diff
@@ -122,3 +123,14 @@ def test_get_cluster_from_options_both_options_invalid():
             {"hosts": {0: {"db": 0}}, "cluster": "foo", "foo": "bar"},
             cluster_manager=manager,
         )
+
+
+def test_construct_redis_cluster_with_is_redis_cluster_flag():
+    cluster = _RedisCluster()
+    lazy_object = cluster.factory(
+        decode_responses=True,
+        is_redis_cluster=True,
+        hosts=[{"host": "127.0.0.1", "port": 6379}],
+    )
+    with pytest.raises(redis.exceptions.RedisClusterException):
+        assert lazy_object
```

Comment:
what's the error being raised here?

Reply:
`assert` initializes the lazy object, and because it is not a reachable node it throws the following error:

```
E   redis.exceptions.RedisClusterException: Redis Cluster cannot be connected. Please provide at least one reachable node: Cluster mode is not enabled on this node
```